lib/mongo/uri.rb

```diff
@@ -256,6 +256,19 @@ def initialize(string, options = {})
       parsed_scheme, _, remaining = string.partition(SCHEME_DELIM)
       raise_invalid_error!(INVALID_SCHEME) unless parsed_scheme == scheme
       parse!(remaining)
+
+      # Warn if conflicting insecure TLS options are provided.
+      if @uri_options[:ssl_verify] == false
+        if @uri_options[:ssl_verify_certificate]
+          log_warn("tlsInsecure is set to disable verification, but tlsAllowInvalidCertificates " +
+                     "is set to enable it")
```

Comment:
Can this say something like "tlsInsecure is set to disable verification, but tlsAllowInvalidCertificates is set to enable it; tlsAllowInvalidCertificates takes precedence" or "Enabling certificate verification due to tlsAllowInvalidCertificates=false setting despite conflicting tlsInsecure=true setting"?